api/clients/disperser_client_v2.go

```diff
@@ -0,0 +1,234 @@
+	payment.CumulativePayment = big.NewInt(0)
+
+	if len(quorums) == 0 {
+		return nil, [32]byte{}, api.NewErrorInvalidArg("quorum numbers must be provided")
+	}
+
+	for _, q := range quorums {
+		if q > corev2.MaxQuorumID {
+			return nil, [32]byte{}, api.NewErrorInvalidArg("quorum number must be less than 256")
+		}
+	}
+
+	// check every 32 bytes of data are within the valid range for a bn254 field element
+	_, err = rs.ToFrArray(data)
+	if err != nil {
+		return nil, [32]byte{}, fmt.Errorf("encountered an error to convert a 32-bytes into a valid field element, please use the correct format where every 32bytes(big-endian) is less than 21888242871839275222246405745257275088548364400416034343698204186575808495617 %w", err)
+	}
+
+	var blobCommitments encoding.BlobCommitments
+	if c.prover == nil {
+		// if prover is not configured, get blob commitments from disperser
+		commitments, err := c.GetBlobCommitment(ctx, data)
+		if err != nil {
+			return nil, [32]byte{}, fmt.Errorf("error getting blob commitments: %w", err)
+		}
+		deserialized, err := encoding.BlobCommitmentsFromProtobuf(commitments.GetBlobCommitment())
+		if err != nil {
+			return nil, [32]byte{}, fmt.Errorf("error deserializing blob commitments: %w", err)
+		}
+		blobCommitments = *deserialized
+	} else {
+		// if prover is configured, get commitments from prover
+		blobCommitments, err = c.prover.GetCommitments(data)
+		if err != nil {
+			return nil, [32]byte{}, fmt.Errorf("error getting blob commitments: %w", err)
+		}
+	}
+
+	blobHeader := &corev2.BlobHeader{
+		BlobVersion:     blobVersion,
+		BlobCommitments: blobCommitments,
+		QuorumNumbers:   quorums,
+		PaymentMetadata: payment,
+	}
+	sig, err := c.signer.SignBlobRequest(blobHeader)
+	if err != nil {
+		return nil, [32]byte{}, fmt.Errorf("error signing blob request: %w", err)
+	}
+	blobHeader.Signature = sig
+	blobHeaderProto, err := blobHeader.ToProtobuf()
+	if err != nil {
+		return nil, [32]byte{}, fmt.Errorf("error converting blob header to protobuf: %w", err)
+	}
+	request := &disperser_rpc.DisperseBlobRequest{
+		Data:       data,
+		BlobHeader: blobHeaderProto,
+	}
+
+	reply, err := c.client.DisperseBlob(ctx, request)
+	if err != nil {
+		return nil, [32]byte{}, fmt.Errorf("error while calling DisperseBlob: %w", err)
+	}
+
+	blobStatus, err := dispv2.BlobStatusFromProtobuf(reply.GetResult())
+	if err != nil {
+		return nil, [32]byte{}, err
+	}
+
+	return &blobStatus, corev2.BlobKey(reply.GetBlobKey()), nil
+}
+
+func (c *disperserClientV2) GetBlobStatus(ctx context.Context, blobKey corev2.BlobKey) (*disperser_rpc.BlobStatusReply, error) {
+	err := c.initOnceGrpcConnection()
+	if err != nil {
+		return nil, api.NewErrorInternal(err.Error())
+	}
+
+	request := &disperser_rpc.BlobStatusRequest{
+		BlobKey: blobKey[:],
+	}
+	return c.client.GetBlobStatus(ctx, request)
+}
+
+func (c *disperserClientV2) GetBlobCommitment(ctx context.Context, data []byte) (*disperser_rpc.BlobCommitmentReply, error) {
```

Comment:
Since this will be more on the public facing side of things, wondering if it's worth it to write godocs.

Reply:
Good call. Done.